lib/routes/account.js

```diff
@@ -845,15 +845,39 @@ module.exports = function (
         var sessionToken = request.auth.credentials
         // Clients have been known to send spurious device updates,
         // which generates lots of unnecessary database load.
-        // Don't write out the update if nothing has actually changed.
-        if (payload.id && sessionToken.deviceId &&
-          payload.id === sessionToken.deviceId.toString('hex') &&
-          (! payload.name || payload.name === sessionToken.deviceName) &&
-          (! payload.type || payload.type === sessionToken.deviceType) &&
-          (! payload.pushCallback || payload.pushCallback === sessionToken.deviceCallbackURL) &&
-          (! payload.pushPublicKey || payload.pushPublicKey === sessionToken.deviceCallbackPublicKey)) {
-          log.info({ op: 'Account.device.spuriousUpdate' })
-          return reply(payload)
+        // Check if anything has actually changed, and log lots metrics on what.
+        if (payload.id) {
+          var spurious = true
+          if(! sessionToken.deviceId || payload.id !== sessionToken.deviceId.toString('hex')) {
+            spurious = false
+            log.increment('device.update.sessionToken')
+          }
+          if (payload.name && payload.name !== sessionToken.deviceName) {
+            spurious = false
+            log.increment('device.update.name')
+          }
+          if (payload.type && payload.type !== sessionToken.deviceType) {
+            spurious = false
+            log.increment('device.update.type')
+          }
+          if (payload.pushCallback && payload.pushCallback !== sessionToken.deviceCallbackURL) {
+            spurious = false
+            log.increment('device.update.pushCallback')
+          }
+          if (payload.pushPublicKey && payload.pushPublicKey !== sessionToken.deviceCallbackPublicKey) {
+            spurious = false
+            log.increment('device.update.pushPublicKey')
+          }
```

Comment:
Purely for readability, is it worth considering extracting the above to, say, an `isSpuriousUpdate` function?